subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/GradleEnterprisePluginExecutionPhaseListener.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise;
+
+/**
+ * Used to signal a start of the execution phase to the plugin.
+ *
+ * Expected to be invoked at most once for a build tree, isn't invoked if the configuration phase fails.
```

Comment:
_isn't invoked if the configuration phase fails._
That's a breaking change for users which won't get their usual stuff they do on the background callback, if the configuration phase fails (like capturing tags, Git status or whatnot)
Is there really no way to call the callback when the configuration phase fails ?

Reply:
It might be easier to handle at the build scan side. A possible implementation might be to have all jobs submitted to wait on a latch. The latch is released when the execution starts or when the build is finished. The second `release` will ensure that jobs will run even if the execution phase never starts.

Reply:
I guess we should also think about how this would work for TAPI model requests without task execution.

Reply:
Build finish trick above should work with the TAPI too, but this might limit potential parallelism. I'm not sure where the line between the configuration phase and everything else lies in the case of TAPI, so I'm not sure what we actually can run in parallel there.

Reply:
After a discussion, it seems that the TAPI lifecycle is too complex to come up with something smarter than `buildFinished`. This PR is intended to be a short-term solution anyway, with the migration to provider-based approach being the next step. The latter would be the one to optimize and polish.

@facewindu, do you think the solution [above](https://github.com/gradle/gradle/pull/20516#discussion_r855195426) is feasible?

Reply:
I think so. Waiting on @ldaley to weigh in

Reply:
I think it will work. We'll need to test it first before confirming.

Reply:
The “start of execution phase” is not a canonically defined epoch in the Gradle model. I suggest we specify precisely what Gradle means by this, at least for this specific usage, here as it's likely to become a point of confusion in the future if the past is any guide.

Reply:
I've expanded the description in the Javadoc to cover included plugin builds, as these are going to be the primary confusion sources.